### PR TITLE
Bids 2859/tootlip list suggestion

### PR DESCRIPTION
**Pull request by MauserBitfly**
This PR:
- creates a component for rendering texts with placeholders
   - highlight: you can have 1-3 highlight placeholders in the translations (these will be bold)
   - url: you can have a url placeholder in the translations
   - you can use any props that you pass in the option as placeholder in the translation
   Point two and three also work in the highlighted placeholders
- create a list component to render a simple text list that uses the text new renderer for it's items   
- create a component that dynamically renders the info icon and it's tooltip if there is something to display in the tooltip. It used the components above to render the tooltip
- Use the new tooltip renderer in the SubscriptionRow.vue

The components are tested with the current use-cases and don't support all different kind of possibilities but should be easily expandable if we need to.

### CODE REVIEW

File: frontend/utils/translation.ts [hunk 1]
```diff
@@ -50,3 +50,11 @@ export function tAll (t: ComposerTranslation, path : string, options?: any) : st
   }
   return list
 }
+
+/** returns an array of translations path's for a list */
```
Comment (marcel-bitfly) [inline frontend/utils/translation.ts:54]: :thinking: does not return an array

File: frontend/components/bc/tooltip/renderer/BcTooltipRendererList.vue [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+    />
+  </ul>
+</template>
+
+<style scoped lang="scss">
+.renderer-list {
+  text-align: left;
+  padding: 0;
+  margin: 0;
+  padding-left: 1.4em;
```
[the remark this answers is not in the document]
Comment (MauserBitfly) [reply frontend/components/bc/tooltip/renderer/BcTooltipRendererList.vue]: right, switched to px (I copied the style from the original)

File: frontend/components/bc/tooltip/renderer/BcTooltipRendererText.vue [hunk 3]
```diff
@@ -0,0 +1,36 @@
+<script setup lang="ts">
+const HIGLIGHT_KEY = ['highlight', 'highlight_1', 'highlight_2', 'highlight_3'] // possible highlight key's in the translation
+defineProps<{
+  tPath: string, // Path to the translation
+  tHighlightPath?: string, // Path to the translation highlight definitions
+  tOptions?: any // translation options
+  tag?: string // html tag to wrap the translations
+  url?: string // url for a url placeholder
+  target?: string // url target - defaults to _blank
```
[the remark this answers is not in the document]
Comment (MauserBitfly) [reply frontend/components/bc/tooltip/renderer/BcTooltipRendererText.vue]: comments removed

File: frontend/components/bc/tooltip/renderer/BcTooltipRendererText.vue [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+}>()
+
+</script>
+
+<template>
+  <i18n-t :keypath="tPath" :tag="tag || 'span'" :plural="tOptions?.plural">
```
[the remark this answers is not in the document]
Comment (MauserBitfly) [reply frontend/components/bc/tooltip/renderer/BcTooltipRendererText.vue:15]: yes some translations do

File: frontend/components/bc/tooltip/renderer/BcTooltipRendererText.vue [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+    <!-- render highlight placeholders -->
+    <template v-for="(hKey, hIndex) in HIGLIGHT_KEY" :key="hKey" #[hKey]>
+      <BcTooltipRendererText
+        v-if="tHighlightPath"
+        class="bold"
+        :t-path="`${tHighlightPath}${hIndex ? `[${hIndex-1}]`:''}`"
+        :t-options="tOptions"
+        :url="url"
+        :target="target"
+      />
```
[the remark this answers is not in the document]
Comment (MauserBitfly) [reply frontend/components/bc/tooltip/renderer/BcTooltipRendererText.vue]: switched to `key`  to keep it short

File: frontend/utils/translation.ts (hunk 1 above)
Comment (MauserBitfly) [reply frontend/utils/translation.ts]: oh damn you copy and paste

File: frontend/components/notifications/management/SubscriptionRow.vue [hunk 6]
```diff
@@ -15,7 +13,7 @@ const { t } = useI18n()
 
 const parentVmodel = defineModel<InternalEntry>({ required: true })
 
-const tooltipLines = computed(() => {
+const tOptions = computed(() => {
```
[the remark this answers is not in the document]
Comment (MauserBitfly) [reply frontend/components/notifications/management/SubscriptionRow.vue]: sure renamed it to `hintTranslationOptions`

**Pull request closed without merging**